exporter/elasticsearchexporter/model.go

```diff
@@ -321,6 +334,137 @@ func (m *encodeModel) upsertMetricDataPointValue(documents map[uint32]objmodel.D
 	return nil
 }
 
+func (m *encodeModel) upsertMetricDataPointValueOTelMode(documents map[uint32]objmodel.Document, resource pcommon.Resource, resourceSchemaURL string, scope pcommon.InstrumentationScope, scopeSchemaURL string, metric pmetric.Metric, dp dataPoint, value pcommon.Value) error {
+	// documents is per-resource. Therefore, there is no need to hash resource attributes
+	hash := metricOTelHash(dp, scope.Attributes(), metric.Unit())
+	var (
+		document objmodel.Document
+		ok       bool
+	)
+	if document, ok = documents[hash]; !ok {
+		document.AddTimestamp("@timestamp", dp.Timestamp())
+		if dp.StartTimestamp() != 0 {
+			document.AddTimestamp("start_timestamp", dp.StartTimestamp())
+		}
+		document.AddString("unit", metric.Unit())
+
+		// At this point the data_stream attributes are expected to be in the record attributes,
+		// updated by the router.
+		// Move them to the top of the document and remove them from the record
+		attributeMap := dp.Attributes()
+
+		forEachDataStreamKey := func(fn func(key string)) {
```

Comment:
Refactoring done in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34472